tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -890,6 +890,36 @@ def raise_soon():
     else:
         raise TestFailure("Exception was not raised")
 
+
+@cocotb.test(expect_error=True)
+def test_nested_exc_traceback(dut):
+    """
+    Test that when an exception is passed up through coroutines
+    the same exception object is raised in the test, and that the
+    full traceback is shown.
+    """
+    the_exception = ValueError("Original exception")
+
+    @cocotb.coroutine
+    def raise_exc():
+        yield Timer(10)
+        raise the_exception
+
+    @cocotb.coroutine
+    def passthrough():
+        yield Timer(10)
+        yield raise_exc()
+
+    try:
+        yield passthrough()
+    except ValueError as e:
+        assert e is the_exception
+        # raise the exception to display traceback
+        raise
```

Comment:
Instead of doing this, can you `traceback.format_exc` the excption, and do some basic string checks?

`(expect_error=True)` will consider failing assertions a pass